File: backend/src/main/java/com/gdje_izlazimo/project/dto/request/create/CreateVenueOperatingHoursRequest.java
package com.gdje_izlazimo.project.dto.request.create;

import com.gdje_izlazimo.project.enums.DayOfWeek;
import jakarta.validation.constraints.NotNull;

import java.time.LocalTime;
import java.util.UUID;

public record CreateVenueOperatingHoursRequest(
        @NotNull(message = "Venue ID is required")
        UUID venueId,
        @NotNull(message = "Start day is required")
        DayOfWeek startDay,
        @NotNull(message = "End day is required")
        DayOfWeek endDay,
        @NotNull(message = "Open time is required")
        LocalTime openTime,
        @NotNull(message = "Closed time is required")
        LocalTime closedTime
) {
}

File: frontend/src/app/core/models/venue-operating-hours/create-venue-operating-hours.request.ts
import { DayOfWeek } from './day-of-week.enum';

export interface CreateVenueOperatingHoursRequest {
  venueId: string;
  startDay: DayOfWeek;
  endDay: DayOfWeek;
  openTime: string;    
  closedTime: string;  
}
